ckan/config/middleware/flask_app.py

```diff
@@ -291,22 +265,22 @@ def save_session(self, app: Any, session: Any, response: Any):
                          'make_error_log_middleware.'
                          .format(plugin.__class__.__name__))
 
-    # Initialize repoze.who
-    who_parser = WhoConfig(conf['here'])
-    who_parser.parse(open(cast(str, conf['who.config_file'])))
-
-    app = PluggableAuthenticationMiddleware(
-        RepozeAdapterMiddleware(app),
-        who_parser.identifiers,
-        who_parser.authenticators,
-        who_parser.challengers,
-        who_parser.mdproviders,
-        who_parser.request_classifier,
-        who_parser.challenge_decider,
-        logging.getLogger('repoze.who'),
-        logging.WARN,  # ignored
-        who_parser.remote_user_key
-    )
+    # Initialize flask-login
+    login_manager = LoginManager()
+    login_manager.init_app(app)
+    # make anonymous_user an instance of CKAN custom class
+    login_manager.anonymous_user = model.AnonymousUser
+    # The name of the view to redirect to when the user needs to log in.
+    login_manager.login_view = "user.login"  # type: ignore
+
+    @login_manager.user_loader
+    def load_user(user_id: str) -> Optional["model.User"]:  # type: ignore
+        return model.User.get(user_id)
```

Comment:
we are firing it on `login()` 
![image](https://user-images.githubusercontent.com/72216462/170493631-b3dc8d70-2a92-480f-9f72-b16fb450ed0b.png)

the purpose of this callback function is:
That user_id will come from the user session. If they're logged in, they will have it set as their identity. When a request comes in, user loader attempts to load a user object using this identity and check whether the user exists and is active (not blocked). If all goes well the user is considered authenticated and the object user loader returns can be accessed across the app (within request context) as current_user.

Reply:
i also removed the extra query here (in def login() ):
<img width="440" alt="image" src="https://user-images.githubusercontent.com/72216462/170498198-5fe8a14e-cdb9-443f-a109-66983bf1dff3.png">

as we have the same logic in `authenticator.ckan_authenticator` and instead of returning `username`, I modifed it to return `userboj` as we need it in order to log the user in via `login_user`